swingset/src/main/java/com/nqadmin/swingset/utils/SSCommon.java

```diff
@@ -395,31 +419,64 @@ public SSCommon(final SSComponentInterface _ssComponent) {
 //	}
 
 	/**
-	 * Methold to add a document listener when the SwingSet component is a
+	 * Method to add a document listener when the SwingSet component is a
 	 * JTextComponent.
 	 */
-	public void addDocumentListener() {
-		((javax.swing.text.JTextComponent) getSSComponent()).getDocument().addDocumentListener(documentListener);
+	private void addSSDocumentListener() {
+		((javax.swing.text.JTextComponent) getSSComponent()).getDocument().addDocumentListener(ssDocumentListener);
 	}
 
 	/**
 	 * Method to add the RowSet listener.
 	 */
 	public void addRowSetListener() {
-		if (!rowsetListenerAdded) {
+		if (rowSetListener==null) {
+			rowSetListener = new SSRowSetListener();
+		}
+		if (!rowSetListenerAdded && rowSet!=null) {
 			rowSet.addRowSetListener(rowSetListener);
-			rowsetListenerAdded = true;
+			rowSetListenerAdded = true;
+			logger.debug(boundColumnName + " - RowSet Listener added.");
 		}
 	}
 	
 	/**
 	 * Method to add any SwingSet Component listener(s).
 	 */
-	public void addSSComponentListener() {
+	public final void addSSComponentListener() {
 		if (!ssComponentListenerAdded) {
-			ssComponent.addSSComponentListener();
-			ssComponentListenerAdded = true;
+			
+			Object eventListener = getSSComponent().getSSComponentListener();
+			
+			if (ssComponent instanceof SSCheckBox) {
+				((SSCheckBox)ssComponent).addItemListener((ItemListener) eventListener);
+			} else if (ssComponent instanceof SSBaseComboBox) {
+				((SSBaseComboBox<?, ?, ?>)ssComponent).addActionListener((ActionListener) eventListener);
+			} else if (ssComponent instanceof SSImage) {
+				((SSImage)ssComponent).getBtnUpdateImage().addActionListener((ActionListener) eventListener);
+			} else if (ssComponent instanceof SSLabel) {
+				((SSLabel)ssComponent).addPropertyChangeListener("text", ((PropertyChangeListener) eventListener));
+			} else if (ssComponent instanceof SSList) {
+				((SSList)ssComponent).addListSelectionListener((ListSelectionListener) eventListener);
+			} else if (ssComponent instanceof SSSlider) {
+				((SSSlider)ssComponent).addChangeListener((ChangeListener) eventListener);
+			} else if (ssComponent instanceof SSFormattedTextField) {
+				((SSFormattedTextField)ssComponent).addPropertyChangeListener("value", ((PropertyChangeListener) eventListener));
+//			} else if (ssComponent instanceof SSTextArea) {
+//			} else if (ssComponent instanceof SSTextField) {
+			} else if (ssComponent instanceof JTextComponent) {
+				addSSDocumentListener();
+			} else {
+				// DIPLAY WARNING FOR UNKNOWN EVENT LISTENER
+				String message = String.format("%s - Encountered unknown Component Event Listener for: %s. Unable to add component listener.",
+						getColumnForLog(), ssComponent.getClass().getSimpleName());
+				logger.error(message);
+				JOptionPane.showMessageDialog((JComponent)getSSComponent(), message, "Unknown Component Event Listener", JOptionPane.ERROR_MESSAGE);
+			}
 		}
```

Comment:
**So if you study one thing carefully in this PR, this method should be that thing. And the removeSSComponentListener() counterpart.**

Reply:
> this method should be that thing. And the removeSSComponentListener() counterpart.

I haven't looked yet, but is that "one thing" `addSSDocumentListener`? And is `removeSSComponentListener` the right name?

Reply:
Looking for a sanity check on SSCommon.addSSComponentListener() and SSCommon.removeSSComponentListener()

Reply:
Good point. I added JavaDoc. Not sure how I would enforce the user to stash a single listener in SSCommon?

Reply:
> enforce the user to stash a single listener in SSCommon?

Do it yourself?
```
private final EventListener eventListener;
public SSCommon(final SSComponentInterface _ssComponent) {
    eventListener = _ssComponent.getSSComponentListener();
```

Reply:
OK. So you'd still have getSSComponentListener(), but it would become single use and the implementation could probably be simplified to return new SSXYZComponentListener();